### PR TITLE
Fixes oc delete ignoring --grace-period.

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/15060
Picks https://github.com/kubernetes/kubernetes/pull/48582

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

**Comment by deads2k** [comment]
@sttts saw you on the upstream

**Comment by fabianofranz** [comment]
flake re[test]

**Comment by fabianofranz** [comment]
[merge][severity: blocker] here and I'll address the minor comments upstream.

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -284,7 +284,7 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 	return nil
 }
 
-func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, shortOutput bool, mapper meta.RESTMapper) error {
+func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, gracePeriod int, shortOutput bool, mapper meta.RESTMapper) error {
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: as in upstream, a godoc would be nice about this int. time.Duration would be obvious, int is not.

**Comment by sttts** [comment]
Looks good to me. Only nit: a godoc wouldn't harm for the int.

**Pull request merged**